pkg/controller/backup_sync_controller.go

```diff
@@ -107,4 +126,36 @@ func (c *backupSyncController) run() {
 			}
 		}
 	}
+
+	c.deleteUnused(cloudBackupNames)
+	return
+}
+
+// deleteUnused deletes backups from Kubernetes if they do not exist in object storage
+func (c *backupSyncController) deleteUnused(cloudBackupNames sets.String) {
+	// Backups from Kubernetes
+	backups, err := c.backupLister.Backups(c.namespace).List(labels.Everything())
+	if err != nil {
+		c.logger.WithError(errors.WithStack(err)).Error("Error listing backup from Kubernetes")
+	}
+	if len(backups) == 0 {
+		return
+	}
+
+	// For each backup item in Kubernetes, verify if has a corresponding backup in the cloud object storage. If not, delete it.
+	for _, backup := range backups {
+		fmt.Println("----> trying to delete: ", backup.Name)
+		if !cloudBackupNames.Has(backup.Name) {
+			err := c.client.Backups(backup.Namespace).Delete(backup.Name, &metav1.DeleteOptions{})
```

Comment:
you can pass nil for the second parameter. you can also collapse this and the next line (`if err := ...; err != nil {...`)